### PR TITLE
Local run in a single Docker container

### PR DESCRIPTION
**Pull request by scherniavsky**
This would enable single-instance automatic runs without docker-compose, e.g.: 
```docker run -ti -v ~/GHE/docker-locust/example:/opt/script -e TARGET_HOST=http://example.com -e LOCUST_FILE=./simple.py -e ROLE=local -e AUTOMATIC=1 -e DURATION=10 -e HATCH_RATE=1 -e USERS=2 docker-locust```

```run --net=host -ti -v ~/GHE/docker-locust/example:/opt/script -e TARGET_HOST=http://localhost -e LOCUST_FILE=./simple.py -e ROLE=local docker-locust```

Needs detailed review.

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -91,20 +91,34 @@ def bootstrap():
...
 
+    elif role == 'local':
+        automatic = convert_str_to_bool(os.getenv('AUTOMATIC', str(False)))
+        logger.info('Automatic run: {auto}'.format(auto=automatic))
+        if not os.environ.get('MASTER_HOST'): os.environ["MASTER_HOST"] = '127.0.0.1'
+        os.environ["ROLE"] = 'master'
+        bootstrap(1)
+        os.environ["ROLE"] = 'slave'
+        bootstrap(1)
+        if automatic:
```
Comment (butomo1989) [inline src/app.py]: if is not needed

File: src/app.py [hunk 2]
```diff
@@ -91,20 +91,34 @@ def bootstrap():
...
+                        logger.error('Attempt: {attempt}. Locust master might not ready yet.'
                                      'Status code: {status}'.format(attempt=_, status=res.status_code))
             except ValueError as v_err:
                 logger.error(v_err)
 
+    elif role == 'local':
+        automatic = convert_str_to_bool(os.getenv('AUTOMATIC', str(False)))
+        logger.info('Automatic run: {auto}'.format(auto=automatic))
+        if not os.environ.get('MASTER_HOST'): os.environ["MASTER_HOST"] = '127.0.0.1'
+        os.environ["ROLE"] = 'master'
```
Comment (butomo1989) [inline src/app.py]: using for loop looks better in my opinion:
`for role in ['master', 'slave', 'controller’]: 
      os.environ['ROLE'] = role  
      bootstrap(1)
`

File: src/app.py (hunk 1 above)
Comment (scherniavsky) [reply src/app.py]: I prefer the process that spins up other processes to only spin up those which are needed. It seems that the ```if automatic:``` condition on line 61 is a workaround for docker-compose that always spins up 3 containers, even if AUTOMATIC=0

**Comment by butomo1989** [comment]
👍

**Comment by scherniavsky** [comment]
:+1:

**Pull request merged**